algebra/src/fft/domain/mod.rs

```diff
@@ -168,12 +173,38 @@ pub trait EvaluationDomain<F: PrimeField>: Debug
             // We compute L(z,tau) = u[i]*ls[i]. 
             // Very misleading notation here.
             u.par_iter_mut().zip(ls).for_each(|(tau_minus_r, l)| {
-                *tau_minus_r = l * tau_minus_r;
+                *tau_minus_r *= l;
             });
             u
         }
     }
 
+    /// Given an index which assumes the first elements of this domain are the elements of
+    /// another (sub)domain with size size_s,
+    /// this returns the actual index into this domain.
+    fn reindex_by_subdomain(&self, other_size: usize, index: usize) -> usize {
```

Comment:
As pointed out in [my comment](https://github.com/HorizenOfficial/ginger-lib/pull/34#discussion_r592218326) for the  FFT pull request, this function is not very elegant.

Reply:
Open an issue please, we will address this in future